### PR TITLE
Add Cron table language

### LINKED ISSUE #6547: Support crontab syntax
This is a super low priority for me. I just want crontab syntax highlighting in my README.md's on GitHub which claims to use this library. If that's true, then thank you. I'm very happy with your work. If not then this is just a common config/code format that you don't support yet.

## Language name
crontab

## URL of example repository
If you mean an example implementation of the syntax highlighting.
https://github.com/vim/vim/blob/6ffcc58be32aa1b337bc839cfe173b68cfde7085/runtime/syntax/crontab.vim
If you mean examples of crontab files I can provide those as needed. I didn't find any comprehensive examples in a moderately quick search.

## URL of syntax highlighting grammar
I'm not aware of a formal highlighting specification. Below are a couple example syntax highlighting implementations.
https://github.com/vim/vim/blob/6ffcc58be32aa1b337bc839cfe173b68cfde7085/runtime/syntax/crontab.vim
https://github.com/textmate/cron.tmbundle/blob/master/Syntaxes/Crontab.tmLanguage < I've never used this one but it's another example

## Most popular extensions
I don't know what is meant by this. I suspect that there are no extensions to the crontab format.

## Detected language
Crontab files are detected as the ``crontab`` filetype by vim/neovim.

### PR DESCRIPTION
**Pull request by Nixinova**
- Resolves #6547

## Description

Adds highlighting for Cron files of file name `crontab`.

## Checklist:

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      - [filename 'crontab'](https://github.com/search?q=path%3A**%2Fcrontab&type=code&p=1): 6.1K+ results
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [okfn-brasil/serenata-de-amor](https://github.com/okfn-brasil/serenata-de-amor/blob/e7aeba78100b1ec96c2a79a85d6f5dfd1a8adaf7/contrib/crontab/crontab#L4)
    - Sample license(s):
      - [MIT](https://github.com/okfn-brasil/serenata-de-amor?tab=MIT-1-ov-file#readme)
  - [x] I have included a syntax highlighting grammar:
    - https://github.com/textmate/cron.tmbundle 
      - (License is custom/nonstandard, but is clearly an exceptionless free license)
    
  - [❌] I have added a color
    - Don't know what to put, there's no logo or company involved.
  - [❌] I have updated the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

**Comment by Alhadis** [comment]
> Don't know what to put, there's no logo or company involved.

Then fallback on some artistic license. 😉 I'd use the colour of a manilla folder. **Rationale:* Tables and scheduled tasks evoke images of dusty old offices with stuff that needs to be done. Hell, [this](https://paintman.co.uk/shop/manilla-bs367-standard-colour/) works: `#ead7ac`.

Otherwise, looks good. TextMate's crontab grammar is pretty solid. I've been using it in Atom without a hitch:

Screen Shot 2024-03-17 at 11 34 15 pm

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -8195,6 +8195,16 @@ Zimpl:
...
   language_id: 411
+crontab:
+  type: data
+  aliases:
+  - cron
+  - cron table
+  filenames:
+  - crontab
+  tm_scope: text.crontab
+  ace_mode: text
```
Comment (Alhadis) [inline lib/linguist/languages.yml:8206]: ACE's Tcl mode does a good job of [highlighting](https://ace.c9.io/build/kitchen-sink.html) <samp>cron(8)</samp> tables:
ACE's Tcl mode highlighting PR's crontab sample
```suggestion
  ace_mode: tcl
```

File: lib/linguist/languages.yml [hunk 2]
```diff
@@ -8195,6 +8195,16 @@ Zimpl:
   tm_scope: none
   ace_mode: text
   language_id: 411
+crontab:
+  type: data
```
Comment (Alhadis) [inline lib/linguist/languages.yml:8199]: The colour I [suggested](https://github.com/github-linguist/linguist/pull/6759#issuecomment-2002443424) earlier:
```suggestion
crontab:
  type: data
  color: "#ead7ac"
```

**Review by Alhadis — CHANGES_REQUESTED** [review]

**Comment by Nixinova** [comment]
```
* linguist.git_submodule.cron.tmbundle
  version: f027bc74d926b[8](https://github.com/github-linguist/linguist/actions/runs/8321971057/job/22769019563?pr=6759#step:6:9)bdc0bfaac22a41[9](https://github.com/github-linguist/linguist/actions/runs/8321971057/job/22769019563?pr=6759#step:6:10)d351bf60edd, filename: /home/runner/work/linguist/linguist/vendor/licenses/git_submodule/cron.tmbundle.dep.yml, license: unknown, allowed: false
    - license needs review: unknown
```
Unsure what to do here. The license of the repo is the following, which is a generic permissive free license:
> Permission to copy, use, modify, sell and distribute this
software is granted. This software is provided "as is" without
express or implied warranty, and with no claim as to its
suitability for any purpose.
What needs to be put as the license for this?

**Comment by Alhadis** [comment]
> Unsure what to do here. The license of the repo is the following, which is a generic permissive free license

This is the usual license TextMate uses for its grammar repositories; it's a [permissive](https://github.com/github-linguist/linguist/blob/6aed0bd564f55c4f2d987c2e6035eda7aea26375/vendor/licenses/git_submodule/c.tmbundle.dep.yml) license that's been reviewed and whitelisted in other grammars hosted by the @textmate org.

> What needs to be put as the license for this?

<samp>cron.tmbundle.dep.yml</samp> should have the following contents:

```yaml
---
name: cron.tmbundle
version: f027bc74d926b8bdc0bfaac22a419d351bf60edd
type: git_submodule
homepage: https://github.com/textmate/cron.tmbundle
license: permissive
licenses:
- sources: README.mdown
  text: "If not otherwise specified (see below), files in this repository fall under
    the following license:\n\n\tPermission to copy, use, modify, sell and distribute
    this\n\tsoftware is granted. This software is provided "as is" without\n\texpress
    or implied warranty, and with no claim as to its\n\tsuitability for any purpose.\n\nAn
    exception is made for files in readable text which contain their own license information,
    or files where an accompanying file exists (in the same directory) with a “-license”
    suffix added to the base-name name of the original file, and an extension of txt,
    html, or similar. For example “tidy” is accompanied by “tidy-license.txt”."
notices: []
```

🤔 …however, I'm not sure why `bundle exec licensed` isn't picking up the license body automatically when it (ostensibly) had no issues with [<samp>c.tmbundle.dep.yml</samp>](https://github.com/github-linguist/linguist/blob/master/vendor/licenses/git_submodule/c.tmbundle.dep.yml).

<sub>@lildude, hulp.</sub>

**Review by Alhadis — APPROVED** [review]

**Comment by lildude** [comment]
The cached file has the correct content, we just need to manually set the license as licensed struggles a bit with so cases like this.

File: vendor/licenses/git_submodule/cron.tmbundle.dep.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+---
+type: git_submodule
+name: cron.tmbundle
+version: f027bc74d926b8bdc0bfaac22a419d351bf60edd
+license: unknown
```
Comment (lildude) [inline vendor/licenses/git_submodule/cron.tmbundle.dep.yml:5]:
```suggestion
license: permissive
```

**Pull request merged**